src/stores/federation.js

```diff
@@ -0,0 +1,180 @@
+ *
+ * @param {string} id store name
+ * @param {State} options.state store state structure
+ */
+export const useFederationStore = defineStore('federation', {
+	state: () => ({
+		pendingShares: {},
+		acceptedShares: {},
+	}),
+
+	actions: {
+		/**
+		 * Fetch pending invitations and keep them in store
+		 *
+		 */
+		async getShares() {
+			try {
+				const response = await getShares()
+				response.data.ocs.data.forEach(item => {
+					if (item.state === FEDERATION.STATE.ACCEPTED) {
+						Vue.set(this.acceptedShares, item.id, item)
+					} else {
+						Vue.set(this.pendingShares, item.id, item)
+					}
+				})
+			} catch (error) {
+				console.error(error)
+			}
+		},
+
+		/**
+		 * Add an invitation from notification to the store.
+		 *
+		 * @param {object} notification notification object
+		 */
+		addInvitationFromNotification(notification) {
+			if (this.pendingShares[notification.objectId]) {
+				return
+			}
+			const [remoteServerUrl, remoteToken] = notification.messageRichParameters.roomName.id.split('::')
+			const { id, name } = notification.messageRichParameters.user1
+			const invitation = {
+				accessToken: null,
+				id: notification.objectId,
+				localRoomId: null,
+				remoteAttendeeId: null,
+				remoteServerUrl,
+				remoteToken,
+				roomName: notification.messageRichParameters.roomName.name,
+				state: FEDERATION.STATE.PENDING,
+				userId: notification.user,
+				inviterCloudId: id + '@' + remoteServerUrl,
+				inviterDisplayName: name,
+			}
+			Vue.set(this.pendingShares, invitation.id, invitation)
+		},
+
+		/**
+		 * Mark an invitation as loading in store.
+		 *
+		 * @param {number} id invitation id
+		 * @param {boolean} value loading state
+		 */
+		markInvitationLoading(id, value) {
+			Vue.set(this.pendingShares[id], 'loading', value)
+		},
+
+		/**
+		 * Mark an invitation as accepted in store.
+		 *
+		 * @param {number} id invitation id
+		 * @param {object} conversation conversation object
+		 */
+		markInvitationAccepted(id, conversation) {
+			if (!this.pendingShares[id]) {
+				return
+			}
+			Vue.delete(this.pendingShares[id], 'loading')
+			Vue.set(this.acceptedShares, id, {
+				...this.pendingShares[id],
+				accessToken: conversation.remoteAccessToken,
+				localRoomId: conversation.id,
+				state: FEDERATION.STATE.ACCEPTED,
+			})
+			Vue.delete(this.pendingShares, id)
+		},
+
+		/**
+		 * Accept an invitation by provided id.
+		 *
+		 * @param {number} id invitation id
+		 * @return {object} conversation to join
+		 */
+		async acceptShare(id) {
+			if (!this.pendingShares[id]) {
+				return
+			}
+			this.markInvitationLoading(id, true)
+			try {
+				const response = await acceptShare(id)
+				this.markInvitationLoading(id, false)
+				this.markInvitationAccepted(id, response.data.ocs.data)
+				return response.data.ocs.data
+			} catch (error) {
+				console.error(error)
+				this.markInvitationLoading(id, false)
+			}
```

Comment:
For error handling, maybe it should be handled where the function were called and use showError. User should be informed when it fails IMO

Reply:
![image](https://github.com/nextcloud/spreed/assets/93392545/b26e5410-e496-46eb-81cf-14ccd5994596)